### DISCUSSION TITLE
Software package management?

### DISCUSSION THREAD

**Discussion by davidroberthoare** [epoptes/epoptes#196] [Q&A] [answered]
Hello :) I know this isn't really a question about Epoptes, but this seems like a crowd that would have some answers, so I'm hopeful someone can point me in the right direction.
I'm in the planning stage of migrating a very-old-mac-lab into a linux-lab at my school. Approx 30 machines.
Epoptes looks like a great package for managing the students' activities, but what system(s) do you all typically use to actually image a lab, and subsequently manage the base system-software and installed packages on all the machines to keep them in sync and up-to-date? How do you most-efficiently roll out a new application to all the machines?
The solutions I've found online so far (Puppet, Chef, Foreman) all look quite heavy for my needs, I think, but maybe they are actually perfect and I just need to learn one of them?
I'd appreciate any pointers or descriptions of the solutions you've found works best for school or small-scale labs - - thank you!
David

**Comment by alkisg** [comment]
Hi @davidroberthoare, we developed and we're using https://ltsp.org exactly for that goal.
It's the simplest way to maintain a computer lab ever. It's like maintaining a single PC.
And as an added benefit, the workstations don't need hard disks, they're all netbooted.

**Comment by davidroberthoare** [comment]
Thank you Alkis :) I'll look into it and maybe test a version. Although I'm
worried that it might not be the best solution for my lab which (I forgot
to mention) I will be using largely for audio/video-production and 3D
animation. The default image for the base plus required software is quite
large (if I was using thick clients) and I fear that as thin clients a
classroom full of simultaneous video-editing just wouldn't be possible...
Will investigate further - thanks again!
David

**Comment by alkisg** [comment, marked as answer]
LTSP is for "fat clients" as we call them. The apps run locally on the clients. They do not run on the server.
The disk image does come from the network. But it's loaded sector by sector on demand. It's the same like using a USB disk. The disk can be e.g. 10 GB and the workstation RAM can be 1 GB and it will still work with no extra RAM usage.

Try it, I think it'll suit you just fine. At most, you might want to use a local /home due to your big audio/video files. You'll still get all the benefits of "maintaining a single PC instead of a computer lab".

**Comment by davidroberthoare** [comment]
Ah - I misunderstood completely! Thank you for your clarification!
David
